src/Evolution/Systems/Cce/Equations.hpp

```diff
@@ -0,0 +1,1010 @@
+                                   tags_needed_for_integrand_computation<
+                                       IntegrandTag, TagsCategory::Temporary>>;
+  using argument_tags = tmpl::append<
+      tags_needed_for_integrand_computation<
+          IntegrandTag, TagsCategory::TagsToSwshDifferentiate>,
+      tags_needed_for_integrand_computation<IntegrandTag,
+                                            TagsCategory::SwshDerivatives>,
+      tags_needed_for_integrand_computation<
+          IntegrandTag, TagsCategory::IndependentOfBondiIntegration>>;
+
+  template <typename... Args>
+  static void apply(Args... args) noexcept {
+    ComputeBondiIntegrand<IntegrandTag>::apply(get(args)...);
+  }
+};
+
+/*!
+ * \brief An interface to ComputeBondiIntegrand which is convenient for passing
+ * around clusters of bondi quantities in `Variables`
+ *
+ * Only a subset of the tags for the system are relevant for each Bondi quantity
+ * tag. In order to determine which quantities are required for a given Bondi
+ * quantity tag, one may use the type alias
+ * `integrand_terms_to_compute_for_bondi_variable`. The
+ * `ComputeBondiIntegrandFromVariables` struct always contains a single
+ * function: `evaluate`, which takes five `Variables`, in order:
+ * - `bondi_integrand_terms` : needs only contain `IntegrandTag`, and functions
+ * as the output of the computation.
+ * - `temporary_cache` : the set of temporary spin-weighted scalars used to
+ *   cache reused intermediate values during individual integrand evaluations.
+ * - `pre_swsh_derivative_bondi_quantities` : The set of Bondi values and
+ * intermediate values which appear as arguments for spin-weighted derivatives,
+ * which are primarily radial (y) derivatives
+ * - `swsh_derivatives` : The set of spin-weighted derivatives used in CCE
+ *   computations
+ * - `values_independent_of_bondi_integration` : The set of values which are
+ * useful during CCE computations and are derived from Bondi quantities or
+ * boundary inputs, but are not updated in between hypersurface steps
+ *
+ * For each individual computation, the list of tags which it accesses from the
+ * input variables can be quite lengthy, so there is the helper type alias
+ * `tags_needed_for_integrand_computation`, which is a tmpl::list of the tags
+ * needed for the computation. The type alias takes two template argument, the
+ * first is the `IntegrandTag` computation which it describes, and the second is
+ * a `TagsCategory` type which describes which of the input variables arguments
+ * it is relevant for. The association between inputs and the `TagsCategory` is:
+ * - `pre_swsh_derivative_bondi_quantities` : `TagsToSwshDifferentiate`
+ * - `swsh_derivatives` : `SwshDerivatives`
+ * - `values_independent_of_bondi_integration` : `IndependentOfBondiIntegration`
+ * - `temporary_cache` : `Temporary`
+ */
+template <typename IntegrandTag>
+struct ComputeBondiIntegrandFromVariables {
```

Comment:
I suggest to move this into test helpers. If we end up needing this in the public interface we can always move it back, but it is not clear to me that we do.

Reply:
I'd still lean towards keeping at as part of the public interface, if only because the CCE/SWSH interfaces are currently working with the design goal of having interfaces which act on clusters of tags with some mathematical and functional commonality, in order to manage the combinatorically large number of values that appear in the calculations and impose some mild hierarchy. It ends up having a nice connection as well to the way the lists of tags are presented in this namespace, where if the tags were all put into a single large list, it would be more difficult to read and maintain.

If there is consensus that the Variables interface should not be included in `src`, I will move it to the tests.

Another alternative is to make the templates more general still, and permit them to take an arbitrary template, so this code is equally operable for sets of Variables or TaggedTuples.

Reply:
I'd like to get input from the other reviews on this. My intention here is to not make the interface unnecessarily complex before we know we need that functionality.

Reply:
Alright, I'm happy to have other input for this one.

My main point is just that while it's conceptually simple to just take a bunch of arguments, it will tend to be a frustrating interface to mechanically use for calling code for the large number of arguments for some of these.

@hrueter @fmahebert @markscheel opinions?

Reply:
Right, so I think we agree that calling these functions with a list of arguments is conceptually nice and clean, but cumbersome when one has to write it out. That is, however, why we have the struct with the `argument_tags` typelist, so that you can just hand the struct over to the DataBox and have it sort out the arguments for you. I suggest you look at the `GeneralizedHarmonic` code and tests for an example. Those are just structs with a plain `argument_tags` list and an `apply` function that takes those arguments.

When we discussed this last week you mentioned you are worried about having to call the functions explicitly in tests, which is why I suggested to move this "short-cut" code with the categories and `Variables` to the tests.

Reply:
From separate discussion, we concluded that it will be best to move these to the test helpers for this PR as suggested by @nilsleiffischer .